File: packages/python/evalforge/scorers/regex.py
import re
from ..models import EvalCase, EvalResult
from .base import Scorer


class RegexScorer(Scorer):
    name = "regex"

    def __init__(self, pattern: str, flags: int = re.IGNORECASE):
        self._re = re.compile(pattern, flags)

    def score(self, case: EvalCase, output: str) -> EvalResult:
        match = bool(self._re.search(output))
        return self._result(case, output, 1.0 if match else 0.0)

File: packages/typescript/src/scorers/regex.ts
import { EvalCase, EvalResult } from "../models";
import { Scorer, makeResult } from "./base";

export class RegexScorer implements Scorer {
  readonly name = "regex";
  private re: RegExp;

  constructor(pattern: string, flags: string = "i") {
    this.re = new RegExp(pattern, flags);
  }

  score(evalCase: EvalCase, output: string): EvalResult {
    return makeResult(evalCase, output, this.re.test(output) ? 1.0 : 0.0);
  }
}
